src/Features/Core/Portable/Shared/Utilities/LogicalStringComparer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Shared.Utilities
+{
+    internal class LogicalStringComparer : IComparer<string>
+    {
+        [DllImport("shlwapi.dll", CharSet = CharSet.Unicode)]
+        private static extern int StrCmpLogicalW(string psz1, string psz2);
+
+        public static readonly IComparer<string> Instance = new LogicalStringComparer();
+
+        private LogicalStringComparer()
+        {
+        }
+
+        public int Compare(string x, string y)
+        {
+            return StrCmpLogicalW(x, y);
```

Comment:
Okay.  In the case we pass `null` does the process AV, corrupt memory, etc ...  Much worse to be passing bad `null` to native code and get random memory problems vs. using a `null` in managed code and an NRE.